components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/util/SupportUtils.java

```diff
@@ -79,4 +101,62 @@ public static String getAuthenticatedUsername() {
         // Get authenticated username from the thread local.
         return PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
     }
+
+    /**
+     * This builds the custom schema for tenants.
+     * @param userManager
+     * @param tenantId
+     * @throws CharonException
+     */
+    public static void buildCustomSchema(UserManager userManager, int tenantId) throws CharonException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Building scim2 custom attribute schema for tenant with Id: "+ tenantId);
+        }
+
+        if (!SCIMCommonUtils.isCustomSchemaEnabled()) {
+            return;
+        }
+        try {
+            if (userManager.getCustomUserSchemaExtension() != null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Scim2 custom attribute schema is found in the UserManager for tenant with Id: " +
+                            "" + tenantId + ". Hence skip building the Extension Builder");
+                }
+                return;
+            }
+
+            try {
+                SCIMCustomSchemaProcessor scimCustomSchemaProcessor = new SCIMCustomSchemaProcessor();
+                List<SCIMCustomAttribute> attributes =
+                        scimCustomSchemaProcessor.getCustomAttributes(IdentityTenantUtil.getTenantDomain(tenantId),
+                                getCustomSchemaURI());
+                AttributeSchema attributeSchema = SCIMCustomSchemaExtensionBuilder.getInstance()
+                        .buildUserCustomSchemaExtension(attributes);
+                SCIMCustomAttributeSchemaCache.getInstance().addSCIMCustomAttributeSchema(tenantId, attributeSchema);
+            } catch (InternalErrorException e) {
+                throw new CharonException("Error while building scim custom schema", e);
+            }
+        } catch (NotImplementedException | BadRequestException | IdentitySCIMException e) {
+            throw new CharonException("Error while building scim custom schema", e);
+        }
+    }
+
+    public static int getTenantId() {
+
+        String tenantDomain = getTenantDomain();
+        if (StringUtils.isNotBlank(tenantDomain)) {
+            return IdentityTenantUtil.getTenantId(tenantDomain);
```

Comment:
AFAIK IdentityTenantUtil.getTenantId already handles the tenantdomain being null case. Please check